src/core/BufferAttribute.js

```diff
@@ -199,15 +200,32 @@ Object.assign( BufferAttribute.prototype, {
 
 	applyMatrix3: function ( m ) {
 
-		for ( let i = 0, l = this.count; i < l; i ++ ) {
+		if ( this.itemSize === 2 ) {
 
-			_vector.x = this.getX( i );
-			_vector.y = this.getY( i );
-			_vector.z = this.getZ( i );
+			for ( let i = 0, l = this.count; i < l; i ++ ) {
 
-			_vector.applyMatrix3( m );
+				_vector2.x = this.getX( i );
+				_vector2.y = this.getY( i );
```

Comment:
Maybe this is better than what we were doing...
```js
_vector2.fromBufferAttribute( this, i );
```

Reply:
Indeed 👍 . Same for `_vector`.

Reply:
I didn't know this method, it seems better.